src/types.jl

```diff
@@ -539,6 +541,18 @@ function System(;
         throw(ArgumentError("there are $(length(atoms)) atoms but $(length(atoms_data)) atom data entries"))
     end
 
+    df = n_dof(D, length(atoms), boundary)
+    if length(constraints) > 0
+        if neighbor_finder == NoNeighborFinder()
+            throw(ArgumentError("Constraints algorithms require neighbor lists."))
```

Comment:
Allow this, since the user may not be providing pairwise interactions at all.

Reply:
You're referring to the Neighbor finder check?